modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ForEachProcessor.java

```diff
@@ -63,24 +63,27 @@ boolean isIgnoreMissing() {
     }
 
     @Override
-    public void execute(IngestDocument ingestDocument) throws Exception {
+    public IngestDocument execute(IngestDocument ingestDocument) throws Exception {
         List<?> values = ingestDocument.getFieldValue(field, List.class, ignoreMissing);
         if (values == null) {
             if (ignoreMissing) {
-                return;
+                return ingestDocument;
             }
             throw new IllegalArgumentException("field [" + field + "] is null, cannot loop over its elements.");
         }
         List<Object> newValues = new ArrayList<>(values.size());
         for (Object value : values) {
             Object previousValue = ingestDocument.getIngestMetadata().put("_value", value);
             try {
-                processor.execute(ingestDocument);
+                if (processor.execute(ingestDocument) == null) {
```

Comment:
technically shouldn't the ingestDocument be chained for each successive execute call?

Reply:
Jup, that would be cleaner now => Will adjust that tomorrow :)